src/index.ts

```diff
@@ -248,21 +251,23 @@ class Compiler {
   }
 
   private validateEnumMembers(rule: JSONSchema.Schema): EnumType {
-    if (!rule.type) rule.type = 'string'
+    if (rule.type === undefined) rule.type = 'string'
```

Comment:
Yes, _if_ `"type" : "boolean"` is defined. But if it isn't defined, it should _not_ default to string. This is how JSON schema works as far as I know.